File: services/boardbox_catalog/src/main/java/com/demoy/boardbox_catalog/configuration/LiquibaseConfig.java
package com.demoy.boardbox_catalog.configuration;

import com.zaxxer.hikari.HikariDataSource;
import liquibase.integration.spring.SpringLiquibase;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.sql.DataSource;

@Configuration
public class LiquibaseConfig {

    @Bean
    public DataSource liquibaseDataSource(@Value("${spring.datasource.url}") String url,
                                          @Value("${spring.datasource.username}") String user,
                                          @Value("${spring.datasource.password}") String pass) {
        HikariDataSource ds = new HikariDataSource();
        ds.setJdbcUrl(url);
        ds.setUsername(user);
        ds.setPassword(pass);
        return ds;
    }

    @Bean
    public SpringLiquibase liquibase(DataSource liquibaseDataSource,
                                     @Value("${spring.liquibase.change-log}") String changelog) {
        SpringLiquibase liq = new SpringLiquibase();
        liq.setDataSource(liquibaseDataSource);
        liq.setChangeLog(changelog);
        liq.setShouldRun(true);
        return liq;
    }
}

File: services/order-service/src/main/kotlin/com/demoy/orderservice/configuration/LiquibaseConfig.kt
package com.demoy.orderservice.configuration

import liquibase.integration.spring.SpringLiquibase
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.autoconfigure.liquibase.LiquibaseProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jdbc.datasource.DriverManagerDataSource
import javax.sql.DataSource

@Configuration
@EnableConfigurationProperties(LiquibaseProperties::class)
class LiquibaseConfig(
    private val liquibaseProperties: LiquibaseProperties,

    @Value("\${spring.datasource.url}")
    private val jdbcUrl: String,

    @Value("\${spring.datasource.username}")
    private val jdbcUser: String,

    @Value("\${spring.datasource.password}")
    private val jdbcPassword: String,

    @Value("\${spring.datasource.driver-class-name:org.postgresql.Driver}")
    private val jdbcDriver: String,

    @Value("\${spring.liquibase.default-schema:}")
    private val liquibaseDefaultSchema: String
) {

    @Bean
    fun liquibase(): SpringLiquibase {
        val ds: DataSource = DriverManagerDataSource().apply {
            setDriverClassName(jdbcDriver)
            url = jdbcUrl
            username = jdbcUser
            password = jdbcPassword
        }

        return SpringLiquibase().apply {
            dataSource = ds
            changeLog = liquibaseProperties.changeLog
            liquibaseProperties.contexts?.let { setContexts(it.joinToString(",")) }
            liquibaseProperties.defaultSchema?.let { defaultSchema = it }
            isDropFirst = liquibaseProperties.isDropFirst
            setShouldRun(liquibaseProperties.isEnabled)
        }
    }
}
